app/controllers/admins/omniauth_callbacks_controller.rb

```diff
@@ -0,0 +1,11 @@
+class Admins::OmniauthCallbacksController < Devise::OmniauthCallbacksController
+  def google_oauth2
+    auth_hash = request.env['omniauth.auth']
+    @admin = AuthorizationService.authorize(auth_hash)
+    unless @admin.nil?
```

Comment:
best practice tries to avoid using `unless - else` because it's harder to understand. It should be `if @admin.present?`